### PR TITLE
topology: use DAI_PERIODS in calculation of DAI buffer size

### PR DESCRIPTION
**Pull request by yongzhi1**
Uses DAI_PERIODS in calculation of DAI buffer size for BUFFER 0 AND 1 as well.
Otherwise, when the DAI_PERIODS value is not DAI_DEFAULT_PERIOD, the following error
could happen.

localhost ~ # alsa_conformance_test -P hw:0,0 -f S32_LE

[  1742766452.727850] ( 36.927082) c0 pga   1.1   ../audio/volume/volume.c:828  ERROR volume_prepare(): sink buffer size 768 is insufficient < 3 * 384
[  1742766463.509099] ( 10.781250) c0 pipe  1.7   ....../pipeline-params.c:310  ERROR pipeline_prepare(): ret = -12, dev->comp.id = 0
[  1742766472.519516] (  9.010416) c0 ipc           src/ipc/handler-ipc3.c:293  ERROR ipc: pipe 1 comp 0 prepare failed -12
[  1742766481.738265] (  9.218750) c0 pipe  1.7   ......./pipeline-graph.c:383  pipe reset

Signed-off-by: Yong Zhi <yong.zhi@intel.com>

### CODE REVIEW

**Comment by yongzhi1** [comment]
cc @Vamshigopal

**Review by dbaluta — APPROVED** [review]

File: tools/topology/topology1/sof/pipe-volume-demux-playback.m4 [hunk 1]
```diff
@@ -69,10 +69,10 @@ W_MUXDEMUX(0, 1, PIPELINE_FORMAT, 2, 2, SCHEDULE_CORE,
 	LIST(`         ', concat(`DEMUX', PIPELINE_ID)))
 
 # Low Latency Buffers
-W_BUFFER(0, COMP_BUFFER_SIZE(2,
+W_BUFFER(0, COMP_BUFFER_SIZE(DAI_PERIODS,
```
Comment (lgirdwood) [inline tools/topology/topology1/sof/pipe-volume-demux-playback.m4:72]: I think the topology is good but the volume control is wrong here.
```c
	if (sinkb->stream.size < dev->ipc_config.periods_sink * sink_period_bytes) {
		comp_err(dev, "volume_prepare(): sink buffer size %d is insufficient < %d * %d",
			 sinkb->stream.size, dev->ipc_config.periods_sink, sink_period_bytes);
		ret = -ENOMEM;
		goto err;
	}
```
volume should only check if the sink buffer can hold 1 period of data.
@singalsu can you check volume and other components for this check. Thanks.

**Review by lgirdwood — CHANGES_REQUESTED** [review]
This is the wrong fix, but it has exposed a bug in volume.

File: tools/topology/topology1/sof/pipe-volume-demux-playback.m4 (hunk 1 above)
Comment (yongzhi1) [reply tools/topology/topology1/sof/pipe-volume-demux-playback.m4:72]: Ack and thanks @lgirdwood @singalsu for review. This issue was reported as P1 for ADL with sof-tgl-max98357a-rt5682 tplg, will close this PR once a formal solution is provided.

Comment (singalsu) [reply tools/topology/topology1/sof/pipe-volume-demux-playback.m4:72]: @lgirdwood Thanks, I'll start to work with fix for all components with same issue.

Comment (singalsu) [reply tools/topology/topology1/sof/pipe-volume-demux-playback.m4:72]: Almost done... I think I'll leave SRC and ASRC for later. They do not in all cases survive with one period sink.

Comment (singalsu) [reply tools/topology/topology1/sof/pipe-volume-demux-playback.m4:72]: @yongzhi1 Could you please try if PR #4659 works the issue with alsa_conformance_test.

**Comment by yongzhi1** [comment]
Thank you @singalsu , shared PR #4659 to team, time to close this one :)

**Pull request closed without merging**